ERCS/erc-7802.md

```diff
@@ -0,0 +1,236 @@
+
+However, each bridge implements a different interface for minting/burning on non-native chains. For example, L1 bridged tokens need to support the `OptimismMintableERC20` specification when bridged (minted) to the OP Stack, but the `IArbToken` interface when bridged to the Arbitrum Stack.
+
+This interface fragmentation is a massive issue in cross-chain communication among chains via third-party bridges or future canonical solutions. At this point, it is clear that every bridge would benefit from a standardized interface for minted/burnt tokens. 
+
+There have been different attempts in the past to standardize token-bridging interfaces. However, third-party providers are also developing cross-chain token frameworks, such as NTT by Wormhole and OFT by LayerZero. Each framework defines its features, like rate limits and fee switches, and implements its mint and burn versions. The resultant interfaces become highly specific, lacking naming conventions and structures.
+
+The proposed interface includes the most relevant and minimal set of actions used by most of these standards. These actions also do not require any governance or owner participation, in contrast, for instance, to set rate limits.
+
+## Specification
+
+This ERC introduces the `IERC7802` interface.
+
+### Interface Identification
+
+The interface identifier for `IERC7802` is **`0x33331994`**, calculated according to [ERC-165](./eip-165.md) as the XOR of the function selectors of the two functions in the interface:
+
+```solidity
+bytes4 constant INTERFACE_ID_IERC7802 = 
+        bytes4(keccak256("crosschainMint(address,uint256)")) ^
+        bytes4(keccak256("crosschainBurn(address,uint256)"));
+```
+
+or via Solidity as 
+
+```solidity
+type(IERC7802).interfaceId
+```
+
+Implementors MUST ensure that the `supportsInterface` method of ERC-165 returns true for this interface ID to indicate support for `IERC7802`.
+
+
+### Methods
+
+**`crosschainMint`**
+
+Mints `_amount` of token to address `_account`. 
+
+This function works as the minting entry point for bridge contracts. Each implementation is responsible for its access control logic.
+
+```solidity
+function crosschainMint(address _account, uint256 _amount) external;
+```
+
+**`crosschainBurn`**
+
+Burns `_amount` of token from address `_account`.
+
+This function works as the burning entry point for bridge contracts. Each implementation is responsible for its access control logic.
+
+```solidity
+function crosschainBurn(address _account, uint256 _amount) external;
+```
+
+### Events
```

Comment:
Adding source/destination chain id & a unique message id to events would make crosschain data more easily usable. This ~may be a hard requirement for analytics, but hard to know without seeing further build out. This is necessary for joining messages across chains, to then see inflows/outflows by chain and routes.

This gap is a main reason why you don't see this data for 3P bridges. Most sites do ~1-3 L2<>L2 bridges, then give up due to the high overhead.

Reply:
+1

Reply:
Hi @MSilb7 !

Thank you for your suggestion. We have discussed your idea internally and agree that it has value. However, we couldn't find a minimalistic set of parameters to add to the event without making the current standard more opinionated:

- **Identifier**: Incorporating an identifier would require defining an encoding standard within the ERC, which we believe is beyond the scope of this proposal.
- **Adding `from/to` and `chainIdFrom/chainIdTo`**: Including these parameters would require changing the inputs of `crosschainMint` and `crosschainBurn` to allow the bridge to pass them. This approach is opinionated because bridges might have multiple destinations or origins, and imposing single input parameters could limit their flexibility. Moreover, event-collisions would be possible.

We see significant value in adding `msg.sender` to the `CrosschainMint` and `CrosschainBurn` events. This addition would allow integrators to identify which bridge interacts with the tokens, and does not affect the interface's neutrality in anyway. 

I believe your idea has the potential to become a standard for bridge events.

Reply:
On `msg.sender`:
- Is the use case of `msg.sender` that you would only look at transactions sent from that address on other chains? Or only function calls (internal txs) initiating from that address?
- Is this already derivable from the transaction / internal transaction? If so then the only benefit is just that you don't need to use traces, you can just use logs.

My overall opinion is like: Without any way to clearly identify how you identify the source burn for any destination mint, it's not useful. The only value it fulfills now is like: I know that this is intended to be a cross-chain transfer, versus a same-chain transfer. But it still gives me no information about where the transfer came from or goes to. Which then opens up other questions, like "How do applications know if the burn/mint is valid?"

Reply:
adding msg.sender to the event would make indexing significantly easier as we can now find all mints/burns associated with a specific bridge.

Reply:
We just added `msg.sender`. It was very easy to implement and can help off-chain agents.